### PR TITLE
Informal taxon group multi select exclude 180597662

### PR DESCRIPTION
**Pull request by MLumme**
Added tree selection for informal taxon group, changes to autocomplete, filter typing due to new filters, renamed some components, and split a common component out of original collections filter to be used by both it and informal group selector.
Feature branch here https://180597662.dev.laji.fi/

### CODE REVIEW

**Review by aorin — CHANGES_REQUESTED** [review]
When I tested the new feature, I got an error message. Steps to reproduce:
1. Go to https://180597662.dev.laji.fi/observation/list
2. Click “All” button under Informal Groups and then click en empty space
3. Click the button with three dots and then “Select groups” button
4. Error message appears

I glanced though the code and didn't find any major problems. I noticed some small style issues that I marked down, mostly about missing types

File: projects/laji/src/app/shared-modules/extended-group-select/extended-group-select.module.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { LajiUiModule } from 'projects/laji-ui/src/public-api';
+import { IucnExtendedGroupSelectComponent } from './iucn-extended-group-select.component';
+import { ObservationExtendedGroupSelectComponent } from './observation-extended-group-select.component';
+import { TreeSelectModule } from '../tree-select/tree-select.module';
+
+@NgModule({
+  imports: [
+    CommonModule,
+    SharedModule,
+    LajiUiModule,
```
[the remark this answers is not in the document]
Comment (MLumme) [reply projects/laji/src/app/shared-modules/extended-group-select/extended-group-select.module.ts:13]: SharedModule is required, as the group-select that is used to render the old informal group selector resides there, but common and lajiui-modules are indeed unnecessary.

**Comment by MLumme** [comment]
These should be fixed now.

**Review by aorin — APPROVED** [review]
Looks good

**Pull request merged**